system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,330 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Translation;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\Commands\Translation\LocalizationFinder\ArrayHelper;
+use Config\App;
+use Locale;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+
+class LocalizationFinder extends BaseCommand
+{
+    protected $group       = 'Translation';
+    protected $name        = 'lang:find';
+    protected $description = 'Find and save available phrases to translate.';
+    protected $usage       = 'lang:find [options]';
+    protected $arguments   = [];
+    protected $options     = [
+        '--locale'   => 'Specify locale (en, ru, etc...) to save files.',
+        '--dir'      => 'Directory for searching for translations (in relation to the APPPATH).',
+        '--show-new' => 'See only new translations as table. Do not write to files.',
+        '--verbose'  => 'Output detailed information.',
+    ];
+
+    /**
+     * Flag for output detailed information
+     */
+    private bool $verbose = false;
+
+    /**
+     * Flag for showing only translations, without saving
+     */
+    private bool $showNew = false;
+
+    private string $languagePath;
+
+    public function run(array $params)
+    {
+        $this->verbose      = array_key_exists('verbose', $params);
+        $this->showNew      = array_key_exists('show-new', $params);
+        $optionLocale       = $params['locale'] ?? null;
+        $optionDir          = $params['dir'] ?? null;
+        $currentLocale      = Locale::getDefault();
+        $currentDir         = APPPATH;
+        $this->languagePath = $currentDir . 'Language';
+
+        if (ENVIRONMENT === 'testing') {
+            $currentDir         = SUPPORTPATH . 'Services/';
+            $this->languagePath = SUPPORTPATH . 'Language';
+        }
+
+        if (is_string($optionLocale)) {
+            if (! in_array($optionLocale, config(App::class)->supportedLocales, true)) {
+                CLI::error(
+                    'Error: "' . $optionLocale . '" is not supported. Supported locales: '
+                    . implode(', ', config(App::class)->supportedLocales)
+                );
+
+                return EXIT_USER_INPUT;
+            }
+
+            $currentLocale = $optionLocale;
+        }
+
+        if (is_string($optionDir)) {
+            $tempCurrentDir = realpath($currentDir . $optionDir);
+
+            if (false === $tempCurrentDir) {
+                CLI::error('Error: Directory must be located in "' . $currentDir . '"');
+
+                return EXIT_USER_INPUT;
+            }
+
```

Comment:
Should you just check `is_dir`? If this is an extension-less file, the realpath would validate.

Reply:
I check that the path is correct, because you can specify something else in the command option `../../dir./../dir/`
And I get a clean path

Reply:
Method `isSubDirectory()` compare path as strings. Any differences (../../) will cause a crash